backend/backend.py

```diff
@@ -36,15 +37,33 @@ def start_video(self) -> None:
         """ Starts the necessary processes to begin displaying camera feed
             video. """
 
-    @abstractmethod
     def on_base_canvas_updated(self, base_canvas: Canvas) -> None:
+        """ Catches any errors that occurs while the base canvas is being
+            updated. """
+        try:
+            self._on_base_canvas_updated(base_canvas)
+        except:
+            self.send_camera_error()
+
+    @abstractmethod
+    def _on_base_canvas_updated(self, base_canvas: Canvas) -> None:
         """ Updates the base canvas which is drawn on the video. Should be
             called whenever the canvas is updated. """
+        raise NotImplementedError("No instructions for base layer found. ")
```

Comment:
@hallgchris Don't these methods need to be implemented for the backend?

Reply:
They do need to be implemented, but what I'm saying is that you don't need to `raise NotImplementedError` because it's actually impossible to create a subclass without implementing this function. This is because of `@abstractmethod`. Give it a try! Create a class inheriting from `Backend` without implementing anything and try to create an instance of it, python will throw a runtime error and not let you

Reply:
Ohhhh righto, I see what you mean now. My bad, that was a misunderstanding